### ISSUE TITLE
Improve parsing of links and locations sharing the same child dictionary

### ISSUE THREAD

**Issue by brynpickering** [calliope-project/calliope#166]
## Problem description
If there is a space in a link definition, that space remains after splitting the link on a comma
`A,B` -> ['A', 'B'], `A, B` -> ['A', ' B']

If there is a space between a list of locations which share the same definition, the space is actually stripped:

```
locations:
    A,B, C, D:
        techs.some_tech:
```
Is equivalent to
```
locations:
    'A':
        techs.some_tech:
    'B':
        techs.some_tech:
    'C':
        techs.some_tech:
    'D':
        techs.some_tech:
```

So there is no consistency. It would certainly be good to strip link strings after splitting on a comma. It would also be good to distinguish between links and locations, not depend on string concatenation for both.

E.g. 
* Use YAML tuple keys: `[A, B]: some_definition`(YAML) -> `{('A', 'B'): some_definition}`(Python)
* Use directional arrows to specify links: `A->B`. Could be used to specify directionality, e.g. `->` for one way, `<->` for bidirectionality.

## Calliope version
0.6.3

**Comment by brynpickering** [comment]
For 0.6.4, I'd say that we should just auto-strip spaces in links on loading a model, matching that with a warning that we've done it. 

For a future version, this should be merged with issue #215 to look at a new way of defining links in general.

**Comment by sjpfenninger** [comment]
Closing in preference of #324
